server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1429,6 +1438,59 @@ export class AgenticChatController implements ChatHandlers {
             this.#llmRequestStartTime = Date.now()
             // Phase 3: Request Execution
             currentRequestInput = sanitizeRequestInput(currentRequestInput)
+
+            if (this.#shouldCompact(currentRequestCount)) {
+                this.#features.logging.info(
+                    `Entering mid-loop compaction at iteration ${iterationCount} with ${currentRequestCount} characters`
+                )
+                this.#telemetryController.emitCompactNudge(
+                    currentRequestCount,
+                    this.#features.runtime.serverInfo.version ?? ''
+                )
+                const messageId = this.#getMessageIdForCompact(uuid())
+                const confirmationResult = this.#processCompactConfirmation(messageId, currentRequestCount)
+                const cachedButtonBlockId = await chatResultStream.writeResultBlock(confirmationResult)
+                await this.waitForCompactApproval(messageId, chatResultStream, cachedButtonBlockId, session)
+
+                // Run compaction
+                const compactionRequestInput = this.#getCompactionRequestInput(session)
+                const compactionResult = await this.#runCompaction(
+                    compactionRequestInput,
+                    session,
+                    metric,
+                    chatResultStream,
+                    tabId,
+                    promptId,
+                    CompactHistoryActionType.Nudge,
+                    session.conversationId,
+                    token,
+                    documentReference
+                )
+
+                if (!compactionResult.success) {
+                    this.#features.logging.error(`Compaction failed: ${compactionResult.error}`)
+                    return compactionResult
+                }
+
+                // Show compaction summary to user before continuing
```

Comment:
can you share a screen recording of the UX?

Reply:
Added

Reply:
I just looked the demo, there is a long pause during conversation streaming where nothing is shown on the UX. We need to fix, we should also consider if we actually want to show the summary for in-loop compaction